### PR TITLE
[Fix] Send anyway alert shown too often

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

**JIRA:** 
- ["Send anyway?" alert to be shown only once through all conversation (align with iOS)](https://wearezeta.atlassian.net/browse/SQSERVICES-483)
- ["Send anyway?" alert shown even LH is already discovered and alert Legal Hold is active was shown](https://wearezeta.atlassian.net/browse/SQSERVICES-487)

### Issues

The "send anyway" confirmation alert is being shown too often. Currently, it is shown the first time the self user sends a message after legal hold is discovered (either by sending that message, or by other means). This happens for every conversation under legal hold. As per the spec, it should only be shown once when legal hold is discovered by sending a message and not again when sending messages in other legal hold conversations.

### Causes

The confusion stems from the three legal hold states defined in the specs: "disabled", "pending approval", and "enabled". I understood the pending state as a transitional state between the disabled and enabled states, whereby the user needed to actively confirm to move to the enabled state. Since each conversation has a legal hold status, the user needed to confirm legal hold per conversation. Furthermore, we were always presenting the alert even if legal hold was already discovered because we did not distinguish where the discovery came from.

### Solutions

The send anyway alert is shown when handling a particular exception that is thrown when trying to send a message. This exception now only thrown when we detect a legal hold device as part of the missing clients. A consequence of this is that the "pending approval" legal hold status is not needed and has therefore been deprecated.

### Testing

Updated existing tests and performed manual tests.
#### APK
[Download build #3573](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3573/artifact/build/artifact/wire-dev-PR3349-3573.apk)
[Download build #3574](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3574/artifact/build/artifact/wire-dev-PR3349-3574.apk)
[Download build #3576](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3576/artifact/build/artifact/wire-dev-PR3349-3576.apk)
[Download build #3589](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3589/artifact/build/artifact/wire-dev-PR3349-3589.apk)

### CODE REVIEW

File: zmessaging/src/main/scala/com/waz/model/ConversationData.scala [hunk 1]
```diff
@@ -207,8 +199,10 @@ object ConversationData {
   final case class LegalHoldStatus(value: Int)
   object LegalHoldStatus {
     val Disabled = LegalHoldStatus(0)
-    val PendingApproval = LegalHoldStatus(1)
     val Enabled = LegalHoldStatus(2)
+
+    @deprecated("'PendingApproval' status is no longer used. Existing occurrences should be treated as 'Enabled'")
+    val PendingApproval = LegalHoldStatus(1)
```
Comment (gizemb) [inline zmessaging/src/main/scala/com/waz/model/ConversationData.scala:205]: is it used in anywhere?

**Review by gizemb — APPROVED** [review]

File: zmessaging/src/main/scala/com/waz/model/ConversationData.scala (hunk 1 above)
Comment (johnxnguyen) [reply zmessaging/src/main/scala/com/waz/model/ConversationData.scala:205]: We could delete this after a few months... I can add a comment.

Comment (gizemb) [reply zmessaging/src/main/scala/com/waz/model/ConversationData.scala:206]: if it exists in someone's DB, we might need to make a DB migration

Comment (johnxnguyen) [reply zmessaging/src/main/scala/com/waz/model/ConversationData.scala:206]: I added a migration.

**Review by gizemb — APPROVED** [review]

**Pull request merged**